velox/dwio/common/tests/OptionsTests.cpp

```diff
@@ -51,3 +55,108 @@ TEST(OptionsTests, testRowNumberColumnInfoInCopy) {
   ASSERT_EQ(rowNumberColumnInfo.insertPosition, rowNumberColumn.insertPosition);
   ASSERT_EQ(rowNumberColumnInfo.name, rowNumberColumn.name);
 }
+
+TEST(OptionsTests, WriterOptionsSerdeTestRoundTripWithAllFields) {
+  Type::registerSerDe();
+  WriterOptions::registerSerDe();
+
+  auto opts = std::make_shared<WriterOptions>();
+
+  // Schema: row<a:bigint>
+  TypePtr schema = ROW({{"a", BIGINT()}});
+  opts->schema = schema;
+  opts->compressionKind = CompressionKind::CompressionKind_ZSTD;
+  opts->serdeParameters = {{"k1", "v1"}, {"k2", "v2"}};
+  opts->sessionTimezoneName = "America/Los_Angeles";
+  opts->adjustTimestampToTimezone = true;
+
+  // Note: We intentionally do NOT set memoryPool, nonReclaimableSection, or the
+  // factory callables because those are not serialized by design.
+
+  // Serialize
+  folly::dynamic serialized = opts->serialize();
+
+  // Basic shape checks on serialized output
+  ASSERT_TRUE(serialized.isObject());
+  // Always present:
+  ASSERT_TRUE(serialized.count("adjustTimestampToTimezone") == 1);
```

Comment:
ASSERT_EQ

However, let's not hard-code serde format in the test. Rather, let's create round-trip test.